src/foam/core/console/AbstractDAOAgent.js

```diff
@@ -37,10 +38,85 @@ foam.CLASS({
     function addSinkToE(e, s) {
       e.add(s);
     },
-    function addToE() {}
+    function addToE() {},
+    function getSink() { this.createSink(); }
   ]
 });
 
+foam.CLASS({
+  package: 'foam.core.console',
+  name: 'AbstractColumnAwareDAOAgent',
+  extends: 'foam.core.console.AbstractDAOAgent',
+
+  imports: [
+    'block',
+    'dao as referenceDAO',
+    'sinkDAO as dao',
+    'sinkUnlimitedDAO as unlimitedDAO',
+    'columns'
+  ],
+
+  properties: [
+    'props',
+    [ 'useProjection', false ]
+  ],
+
+  methods: [
+    function getSink() {
+      return this.useProjection ? this.getProjectionSink() : this.createSink();
```

Comment:
Or, just keep this as createSink() and let the user have their own this.useProjection ? ..., that seems like the simplest and easiest to understand.